catalog/schematron/catalog-schematron-plugin/src/main/java/ddf/services/schematron/SchematronValidationService.java

```diff
@@ -232,24 +254,28 @@ public SchematronReport getSchematronReport() {
 
     @Override
     public void validate(Metacard metacard) throws ValidationException {
-        for (Templates validator : validators) {
-            String metadata = metacard.getMetadata();
-            if (StringUtils.isEmpty(metadata) || (namespace != null
-                    && !namespace.equals(XMLUtils.getRootNamespace(metadata)))) {
-                return;
-            }
-            schematronReport = generateReport(metadata, validator);
-            if (!schematronReport.isValid(suppressWarnings)) {
-                throw new SchematronValidationException("Schematron validation failed.",
-                        schematronReport.getErrors()
-                                .stream()
-                                .map(SchematronValidationService::sanitize)
-                                .collect(Collectors.toList()),
-                        schematronReport.getWarnings()
-                                .stream()
-                                .map(SchematronValidationService::sanitize)
-                                .collect(Collectors.toList()));
+        try {
+            for (Future<Templates> validator : validators) {
+                String metadata = metacard.getMetadata();
+                if (StringUtils.isEmpty(metadata) || (namespace != null && !namespace.equals(
```

Comment:
Likewise this test should be pulled out of the loop as it doesn't require the validator.